src/gui/main.cpp

```diff
@@ -249,55 +259,14 @@ QSettings::setDefaultFormat(CkbSettings::Format);
     if(tmpSettings->value("Program/HiDPI", false).toBool())
         QApplication::setAttribute(Qt::AA_EnableHighDpiScaling);
 #endif
-
-    // Setup main application
-    QApplication a(argc, argv);
-
-    // Setup translations
-    QTranslator translator;
-    if(translator.load(QLocale(), "", "", ":/translations"))
-        a.installTranslator(&translator);
-
-    const quint16 currentSettingsVersion = tmpSettings->value("Program/SettingsVersion", 0).toInt();
-    if(currentSettingsVersion && currentSettingsVersion > CKB_NEXT_SETTINGS_VER){
-        if(QMessageBox::warning(nullptr, QObject::tr("Downgrade Warning"),
-                                QObject::tr("Downgrading ckb-next will lead to profile data loss. "
-                                            "It is recommended to click Cancel and update to the latest version.<br><br>"
-                                            "If you wish to continue, back up the settings file located at<blockquote>%1</blockquote>and click OK.").arg(tmpSettings->fileName()),
-                                QMessageBox::Ok, QMessageBox::Cancel)
-                == QMessageBox::Cancel){
-            delete tmpSettings;
-            tmpSettings = nullptr;
-            return 0;
-        }
-        // Only handle the downgrade here. The upgrade is handled in CkbSettings.
-        tmpSettings->setValue("Program/SettingsVersion", CKB_NEXT_SETTINGS_VER);
-    }
-
-    // Setup names and versions
-    QCoreApplication::setOrganizationName("ckb-next");
-    QCoreApplication::setApplicationVersion(CKB_NEXT_VERSION_STR);
-    QCoreApplication::setApplicationName("ckb-next");
-
     // Setup argument parser
     QCommandLineParser parser;
     QString errorMessage;
+    bool background = false, daemon = false;
+    {
+      QCoreApplication parser_app(argc, argv);
     parser.setApplicationDescription("Open Source Corsair Input Device Driver for Linux and OSX.");
```

Comment:
If you want, please rebase it against the latest changes, which add CKB_NEXT_DESCRIPTION here.